### DISCUSSION TITLE
500: Error getting user profile from external provider

### DISCUSSION THREAD

**Discussion by CtrlAltDeletedDev** [supabase/supabase#27276] [Questions] [answered]
Hello, first time posting in a Github Discussion ever.

I'm trying to set up just Github OAuth login in my NextJS app and I keep getting a http://localhost:3000/api/auth/callback?error=server_error&error_code=500&error_description=Error+getting+user+profile+from+external+provider#error=server_error&error_code=500&error_description=Error+getting+user+profile+from+external+provider returned.

My Supabase logs indicate a different error from Github about not being able to list email addresses.

{
  "event_message": "{\"component\":\"api\",\"error\":\"403: {\\\"message\\\":\\\"Resource not accessible by integration\\\",\\\"documentation_url\\\":\\\"https://docs.github.com/rest/users/emails#list-email-addresses-for-the-authenticated-user\\\",\\\"status\\\":\\\"403\\\"}\",\"level\":\"error\",\"method\":\"GET\",\"msg\":\"500: Error getting user profile from external provider\",\"path\":\"/callback\",\"referer\":\"http://localhost:3000/\",\"remote_addr\":\"66.42.28.145\",\"request_id\":\"893c6207b408a56c-MIA\",\"time\":\"2024-06-14T18:28:56Z\"}",
  "id": "979cdf67-5f93-4f60-b421-9425f2856837",
  "metadata": [
    {
      "message": null,
      "timestamp": null,
      "__MONOTONIC_TIMESTAMP": null,
      "CODE_FUNC": null,
      "instance_id": null,
      "status": null,
      "_CMDLINE": null,
      "method": "GET",
      "_SYSTEMD_CGROUP": null,
      "CODE_FILE": null,
      "EXECUTABLE": null,
      "_EXE": null,
      "UNIT": null,
      "level": "error",
      "_COMM": null,
      "duration": null,
      "issuer": null,
      "_LINE_BREAK": null,
      "_SOURCE_REALTIME_TIMESTAMP": null,
      "msg": "500: Error getting user profile from external provider",
      "action": null,
      "login_method": null,
      "_UID": null,
      "host": "db-yvxxlqqiihmrbllvtbwr",
      "PRIORITY": null,
      "_CAP_EFFECTIVE": null,
      "_PID": null,
      "INVOCATION_ID": null,
      "_SYSTEMD_UNIT": null,
      "source_type": null,
      "SYSLOG_FACILITY": null,
      "request_id": "893c6207b408a56c-MIA",
      "CODE_LINE": null,
      "path": "/callback",
      "component": "api",
      "project": null,
      "user_id": null,
      "auth_event": [],
      "args": [],
      "referer": "http://localhost:3000/",
      "factor_id": null,
      "provider": null,
      "client_id": null,
      "remote_addr": "yoink",
      "_SYSTEMD_SLICE": null,
      "_SYSTEMD_INVOCATION_ID": null,
      "header": null,
      "_MACHINE_ID": null,
      "_AUDIT_LOGINUID": null,
      "_TRANSPORT": null,
      "_SELINUX_CONTEXT": null,
      "MESSAGE_ID": null,
      "__REALTIME_TIMESTAMP": null,
      "metadata": [],
      "_STREAM_ID": null,
      "metering": null,
      "time": null,
      "_GID": null,
      "_BOOT_ID": null,
      "SYSLOG_IDENTIFIER": null,
      "_AUDIT_SESSION": null,
      "error": "403: {\"message\":\"Resource not accessible by integration\",\"documentation_url\":\"https://docs.github.com/rest/users/emails#list-email-addresses-for-the-authenticated-user\",\"status\":\"403\"}"
    }
  ],
  "timestamp": 1718389736000000
}

I've searched everywhere and came across several Discussions linked to LinkedIn Auth issues but nothing for Github so I'm assuming the issue is on my end.

Here is my GitHub login action

```ts
import { createClient } from "@/lib/supabase/client";
import { redirect } from "next/navigation";

export async function signInWithGithub() {
  const supabase = createClient();

  const { data, error } = await supabase.auth.signInWithOAuth({
    provider: "github",
    options: {
      redirectTo: process.env.NEXT_PUBLIC_SUPABASE_AUTH_CALLBACK,
    },
  });

  if (error) {
    console.error(error);
  }

  if (data.url) {
    redirect(data.url); // use the redirect API for your server framework
  }
}
```

And here is my route

```ts
import { cookies } from "next/headers";
import { NextResponse } from "next/server";
import { type CookieOptions, createServerClient } from "@supabase/ssr";

export async function GET(request: Request) {
  const { searchParams, origin } = new URL(request.url);
  const code = searchParams.get("code");

  console.log(code);
  // if "next" is in param, use it as the redirect URL
  const next = searchParams.get("next") ?? "/guestbook";

  if (code) {
    const cookieStore = cookies();
    const supabase = createServerClient(
      process.env.NEXT_PUBLIC_SUPABASE_URL!,
      process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
      {
        cookies: {
          get(name: string) {
            return cookieStore.get(name)?.value;
          },
          set(name: string, value: string, options: CookieOptions) {
            cookieStore.set({ name, value, ...options });
          },
          remove(name: string, options: CookieOptions) {
            cookieStore.delete({ name, ...options });
          },
        },
      }
    );
    const { error } = await supabase.auth.exchangeCodeForSession(code);
    if (!error) {
      return NextResponse.redirect(`${origin}${next}`);
    }
  }

  // return the user to an error page with instructions
  return NextResponse.redirect(`${origin}/auth/auth-code-error`);
}
```

Any help or guidance is appreciated!

**Comment by Aoniq** [comment]
Are you trying to login with the same user account as you created the github oauth with? I happened to have the same issue with Spotify Oauth login using supabase and I forgot to add my second spotify account to the user management of spotify developers.

**Reply by CtrlAltDeletedDev** [reply]
Yeah I should be. I haven't ever had another account, just the one. But how can I see what it's trying to log me in as to check?

**Comment by engageintellect** [comment]
same issue, using sveltekit:

```
http://localhost:5173/auth/callback?error=server_error&error_code=500&error_description=Error+getting+user+profile+from+external+provider
```

**Comment by CtrlAltDeletedDev** [comment, marked as answer]
My issue was that I did not have proper permissions granted to my App. 

In your GitHub App:

1. On the left sidebar, select **Permissions & events**
2. **Account permissions > Email addresses**
3. Set to Read-only
4. Save

Supabase requires this permission in order to view the user data. Once I did this (and made some other smaller changes that were my own code issues), it resolved it and the user was properly being logged.

**Reply by Balastrong** [reply]
Same issue, this worked for me. Thanks!

**Reply by azizbecha** [reply]
You are a live saver! Been stuck on this issue for 2 days. Thanksss!!

**Reply by notpritamm** [reply]
worked for me thanku

**Reply by rmadrazo97** [reply]
this should be added to @supabase-automated-user 's docs

**Reply by jejrthompson** [reply]
Thank you! I would have never guessed this.

**Reply by zettalyst** [reply]
It saved my time thanks

**Comment by nNEWBE** [comment]
You saved my day 🎉
